Kuzzle.Tests/API/Controllers/RealtimeControllerTest.cs

```diff
@@ -0,0 +1,163 @@
+            });
+            Assert.Equal<Int32>(count, res);
+        }
+
+        [Fact]
+        public async void PublishAsyncTest()
+        {
+            string index = "an_index";
+            string collection = "to";
+            JObject message = new JObject { { "infinity", "and beyond" } };
+
+            await _realtimeController.PublishAsync(index, collection, message);
+
+            _api.Verify(new JObject{
+                {"index" , index },
+                {"collection" , collection },
+                {"controller" , "realtime" },
+                {"action" ,"publish" },
+                {"body" , message },
+            });
+        }
+
+        [Theory]
+        [
+            MemberData(nameof(MockGenerators.GenerateSubscribeOptions),
+            MemberType = typeof(MockGenerators))
+        ]
+        public async void SubscribeAsyncTest(SubscribeOptions options)
+        {
+            string roomId = "A113";
+            string channel = "a_channel";
+            _api.SetResult(new JObject { {
+                "result", new JObject {
+                    { "roomId", roomId },
+                    { "channel", channel } }
+                } }
+            );
+            string index = "an_index";
+            string collection = "a_collection";
+            JObject filters = new JObject { { "studio", "pixar" } };
+            JObject expectedQuery = new JObject {
+                {"index", index},
+                {"collection",collection},
+                {"controller", "realtime"},
+                {"action", "subscribe"},
+                {"body", filters}
+            };
+            if (options != null)
+            {
+                expectedQuery.Merge(JObject.FromObject(options));
+            }
+            Mock<RealtimeController.NotificationHandler> mockNotifHand = new Mock<RealtimeController.NotificationHandler>();
+
+            string res = await _realtimeController.SubscribeAsync(index, collection, filters, mockNotifHand.Object, options);
+
+            _api.Verify(expectedQuery);
+            Assert.Equal(roomId, res);
+        }
+
+        [Fact]
+        public async void UnsubscribeAsyncTest()
+        {
+            //First we need to subscribe to "a_collection"
+            string index = "an_index";
+            string collection = "a_collection";
+            JObject filters = new JObject { { "studio", "pixar" } };
+            string roomId = "A113";
+            string channel = "a_channel";
+            _api.SetResult(new JObject { {
+                "result", new JObject {
+                    { "roomId", roomId },
+                    { "channel", channel } }
+                } }
+            );
+            Mock<RealtimeController.NotificationHandler> mockNotifHand = new Mock<RealtimeController.NotificationHandler>();
+            await _realtimeController.SubscribeAsync(index, collection, filters, mockNotifHand.Object);
+
+            //Then we can test if unsubcription is working
```

Comment:
Not really: you only check that the method sends the correct request, but you do not check that we correctly dereferenced the subscription.

You can do that by either sending a notification to that channel and verify that the mocked `NotificationHandler` is not invoked, and/or by checking directly the content of the rooms cache maintained by the controller object.